srcmedia/ts/components/ActiveFilters.ts

```diff
@@ -0,0 +1,271 @@
+            }
+
+            // for checkboxes we need to fake both change and input events
+            $button.addEventListener('click', () => {
+                $input.checked = false
+                $input.dispatchEvent(new Event('change', { bubbles: true }))
+                $input.dispatchEvent(new Event('input', { bubbles: true }))
+                // this.remove($button)
+            })
+        }
+    }
+
+    /**
+     * Handle events originating from the form itself, e.g. the user typing into
+     * a range filter or clicking a checkbox.
+     *
+     * @param event form events
+     */
+    protected handleFormInput(event: InputEvent): void {
+        const target = event.target as HTMLInputElement
+
+        // determine the type of filter that is affected, then delegate to the
+        // add(), remove(), or change() functions.
+
+        // boolean filters and text facets
+        if (target.type == 'checkbox') {
+            if (target.checked) {           // checked = add new filter
+                this.add({
+                    type: 'checkbox',
+                    input: target.id,
+                    name: target.name,
+                    value: target.value
+                })
+            }
+            else this.remove({              // unchecked = remove filter
+                type: 'checkbox',
+                input: target.id
+            })
+        }
+
+        // range filters
+        else if (target.type == 'number') {
+            
+            // find both inputs via parent fieldset
+            const idMatch = target.id.match(this.rangeFilterPattern) as RegExpMatchArray
+            const $fieldset = document.getElementById(idMatch[1]) as HTMLFieldSetElement
+            const [$rangeStart, $rangeStop]: HTMLInputElement[] = Array.from($fieldset.querySelectorAll('input'))
+            
+            // if both inputs are empty, delete button
+            if ($rangeStart.value == '' && $rangeStop.value == '')
+                this.remove({
+                    type: 'range',
+                    input: idMatch[1]
+                })
+
+            // otherwise create/update existing button
+            else {
+                // generate inner text using both values + legend
+                const $legend = $fieldset.querySelector('legend') as HTMLLegendElement
+                const buttonContent = `${$rangeStart.value} – ${$rangeStop.value}`
+
+                // store the button using the fieldset's id
+                this.update({
+                    type: 'range',
+                    input: idMatch[1],
+                    name: $legend.textContent as string,
+                    value: buttonContent
+                })
+            }
+        }
+    }
+
+    protected buttonText(filter: Filter) {
+        switch(filter.type) {
+            case 'range':
+                return `${filter.name}: ${filter.value}`
+            case 'checkbox':
+            default:
+                return filter.name
```

Comment:
It seems like it would be more logical to use the label for the element rather than the name — in our case, they happen to be the same, except that our facet labels include counts in spans that you'd have to strip out.

Does using the label help any with the card filter?

Reply:
the card filter has a label, but it actually contains no text - just an inner `<span>` whose content says "card". I think "has card" is actually more informative, and that's what in the `name` (although I'd need to urldecode it).

Reply:
expanding on this a little: you can get a reference to any `<input>`'s associated label(s), but in this case the computed content includes the counts, as you noted:
```js
> input.labels[0].innerText
"Card 645"
```

Reply:
looks like I actually have no choice but to use the label, because using `name` doesn't work for the case of empty-valued choices like "unidentified" for gender. evidently it will have to work this way:
```js
> input.labels[0].childNodes[0].textContent.trim()
"Unidentified"
```
where we assume that the `<span>` is not the first direct child of the label, which I think is fair for now.

Reply:
Ah — yes that makes sense. Sounds reasonable to me.